src/Plugin/Queue/Models/Factories/ShowFactory.php

```diff
@@ -0,0 +1,180 @@
+<?php declare(strict_types=1);
+
+/*
+  Copyright (c) 2023, Manticore Software LTD (https://manticoresearch.com)
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License version 2 or any later
+  version. You should have received a copy of the GPL license along with this
+  program; if you did not, you can find it at http://www.gnu.org/
+*/
+
+
+namespace Manticoresearch\Buddy\Base\Plugin\Queue\Models\Factories;
+
+use Manticoresearch\Buddy\Base\Plugin\Queue\Models\Model;
+use Manticoresearch\Buddy\Base\Plugin\Queue\Models\Show\ShowMaterializedViewModel;
+use Manticoresearch\Buddy\Base\Plugin\Queue\Models\Show\ShowMaterializedViewsModel;
+use Manticoresearch\Buddy\Base\Plugin\Queue\Models\Show\ShowSourceModel;
+use Manticoresearch\Buddy\Base\Plugin\Queue\Models\Show\ShowSourcesModel;
+use Manticoresearch\Buddy\Base\Plugin\Queue\Payload;
+
+class ShowFactory {
+
+	/**
+	 * @template T of array
+	 * @phpstan-param T $parsedPayload
+	 * @phpstan-return Model<T>|null
+	 */
+
+	public static function create(array $parsedPayload): ?Model {
+
```

Comment:
Having too many blank lines makes the code harder to read. It's better not to add them when they're unnecessary, like after an opening curly brace { and before a closing curly brace }. Instead, try splitting the code into semantic blocks - that's a better way to organize it.